### DISCUSSION TITLE
[Support]: jsmpeg color offset - EmpireTech IPC-Color4K-T180

### DISCUSSION THREAD

**Discussion by brianfeucht** [blakeblackshear/frigate#13092] [Config Support] [answered]
### Describe the problem you are having

The output from jsmpeg is offset colors

Stream output from jsmpeg in app:
![image](https://github.com/user-attachments/assets/2a19e64a-43f2-49df-862f-02b83c9aaa71)

Stream output in VLC:
![image](https://github.com/user-attachments/assets/3b7b74df-8adb-4ca5-a099-f961376eaa66)

### Steps to reproduce

View this camera via Live View in app

### Version

0.14.0

### Frigate config file

```yaml
# yaml-language-server: $schema=http://192.168.15.35:5000/api/config/schema.json
mqtt:
  host: 192.168.1.1
  user: ###
  password: ###

ffmpeg:
  hwaccel_args: preset-vaapi
  output_args:
    record: preset-record-generic-audio-aac

detectors:
  coral:
    type: edgetpu
    device: usb

objects:
  track:
    - person
    - car
    - cat
    - dog
    - bicycle
    - motorcycle
    - cow
    - horse
    - sheep
    - skis
    - snowboard
    - bird

cameras:
  WestGate:
    webui_url: http://192.168.1.6/
    ffmpeg:
      inputs:
        - path: rtsp://##:###@192.168.1.6/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
        - path: rtsp://##:###@192.168.1.6/cam/realmonitor?channel=1&subtype=2
          roles:
            - detect
    audio:
      enabled: false
    motion:
      mask:
        - 1725,75,1840,75,1840,29,1725,29

record:
  enabled: true
  retain:
    days: 7
    mode: all
  events:
    retain:
      default: 14
      mode: active_objects
      objects:
        person: 30

snapshots:
  enabled: true
  retain:
    default: 30

birdseye:
  enabled: false

tls:
  enabled: false

auth:
  enabled: true

version: 0.14
```

### Relevant log output

```shell
2024-08-15 14:11:55.046729946  [INFO] Preparing Frigate...
2024-08-15 14:11:55.065941619  [INFO] Starting Frigate...
2024-08-15 14:11:56.533356461  [2024-08-15 14:11:56] frigate.app                    INFO    : Starting Frigate (0.14.0-da913d8)
2024-08-15 14:11:56.533432256  [2024-08-15 14:11:56] frigate.util.config            INFO    : Checking if frigate config needs migration...
2024-08-15 14:11:56.556806121  [2024-08-15 14:11:56] frigate.util.config            INFO    : frigate config does not need migration...
2024-08-15 14:12:09.455851944  [2024-08-15 14:12:09] peewee_migrate.logs            INFO    : Starting migrations
2024-08-15 14:12:09.456127218  [2024-08-15 14:12:09] peewee_migrate.logs            INFO    : There is nothing to migrate
2024-08-15 14:12:09.481118740  [2024-08-15 14:12:09] frigate.app                    INFO    : Recording process started: 344
2024-08-15 14:12:09.490605814  [2024-08-15 14:12:09] frigate.app                    INFO    : Recording process started: 350
2024-08-15 14:12:09.492323972  [2024-08-15 14:12:09] frigate.app                    INFO    : go2rtc process pid: 96
2024-08-15 14:12:09.514856680  [2024-08-15 14:12:09] detector.coral                 INFO    : Starting detection process: 373
2024-08-15 14:12:12.135385839  [2024-08-15 14:12:09] frigate.app                    INFO    : Output process started: 375
2024-08-15 14:12:12.135965125  [2024-08-15 14:12:09] frigate.app                    INFO    : Camera processor started for WestGate: 403
2024-08-15 14:12:12.136321165  [2024-08-15 14:12:09] frigate.app                    INFO    : Capture process started for WestGate: 411
2024-08-15 14:12:12.136946678  [2024-08-15 14:12:09] frigate.app                    INFO    : Audio process started: 425
2024-08-15 14:12:12.143824881  [2024-08-15 14:12:09] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2024-08-15 14:12:12.144372471  [2024-08-15 14:12:12] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
```

### Operating system

Debian

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

EmpireTech IPC-Color4K-T180

**Comment by brianfeucht** [comment]
Codec details from VLC
![image](https://github.com/user-attachments/assets/a1748285-6285-455e-aae0-794b158ad10a)

**Comment by hawkeye217** [comment]
This is not a bug. See the [FAQ](https://github.com/blakeblackshear/frigate/discussions/12724), Usage question 15.

You need to set your `detect` width and height manually, or better yet, use go2rtc for your live views.

**Comment by brianfeucht** [comment]
Thanks! I didn't see this release FAQ.   I was looking on the [doc page](https://docs.frigate.video/troubleshooting/faqs) =) 

I've updated the camera config to:
```
cameras:
  WestGate:
    webui_url: http://192.168.1.6/
    ffmpeg:
      inputs:
        - path: rtsp://##:###@192.168.1.6/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
        - path: rtsp://##:###@192.168.1.6/cam/realmonitor?channel=1&subtype=2
          roles:
            - detect
    audio:
      enabled: false
    motion:
      mask:
        - 1725,75,1840,75,1840,29,1725,29
    detect:
      width: 1920
      height: 832
      fps: 5
 ```
 
 Still seeing the same issue.  I'll try switching live view to go2rtc later today.

**Reply by hawkeye217** [reply]
1920x832 is an odd resolution. What happens if you change it to 1920x843 or 1920x844?

**Reply by brianfeucht** [reply]
It is a dual lens camera that stiches the two together
![image](https://github.com/user-attachments/assets/530f6d38-79ec-4411-b394-9351e7edf044)

I switch the live and detect feeds to a go2rtc stream and it resolved the issue.  

Happy to switch back if you'd like to troubleshoot further.

**Reply by hawkeye217** [reply]
Ah, that would make sense then. No need to switch back. Go2rtc is the recommendation over jsmpeg anyways.

**Comment by brianfeucht** [comment]
Switching to a go2rtc feed resolved this for me

**Reply by ChirpyTurnip** [reply]
Hi @brianfeucht!

I think I have the exact same camera (180 degree Dahua / EmpireTech), so I am stuck with exactly the same problem. It's good to know switching to go2rtc fixed it for you but I'm having a few issues as I'm running two instances of Frigate on Unraid for two different camera installations. Sometimes I have conflicts between the two setups...so troubleshooting can be really hard. 

For those of us similarly stuck would you mind sharing your updated config? If I can see a config that should definitely work with my hardware that would be super helpful!

Thanks!

CP.

**Reply by brianfeucht** [reply, marked as answer]
@ChirpyTurnip yes, happy to share 😄 

On the camera I have sub stream 1 disabled as it doesn't provide the resolution I want for detection.  If you want audio detection, be sure to enable audio on substream 2 (it is disabled by default)

Camera setup
![image](https://github.com/user-attachments/assets/3493fe45-66e1-4075-80f3-972aa87b460b)
![image](https://github.com/user-attachments/assets/0c7fab46-d254-412d-a325-503322b53cf4)


```
go2rtc:
  streams:
    west_gate:
      - rtsp://user:pass@192.168.1.6/cam/realmonitor?channel=1&subtype=0
    west_gate_sub:
      - rtsp://user:pass@192.168.1.6/cam/realmonitor?channel=1&subtype=2
cameras:      
  WestGate:
      webui_url: http://192.168.1.6/
      ffmpeg:
        inputs:
          - path: rtsp://127.0.0.1:8554/west_gate
            input_args: preset-rtsp-restream
            roles:
              - record
          - path: rtsp://127.0.0.1:8554/west_gate_sub
            input_args: preset-rtsp-restream
            roles:
              - detect
              - audio
      audio:
        enabled: true
      motion:
        mask:
          - 1725,75,1840,75,1840,29,1725,29
      detect:
        width: 1920
        height: 832
        fps: 5
      live:
        stream_name: west_gate_sub
```

**Reply by ChirpyTurnip** [reply]
That was perfect! I still have some other issues to sort out, but the cameras are now all connected, with sound, and no color distortions! So good progress thank you!
